pkg/sql/drop_type.go

```diff
@@ -194,6 +194,44 @@ func (p *planner) removeTypeBackReferences(
 	return nil
 }
 
+func (p *planner) addRoutineViewBackReference(
+	ctx context.Context, routineID descpb.ID, ref descpb.ID,
+) error {
+	mutDesc, err := p.Descriptors().MutableByID(p.txn).Function(ctx, routineID)
+	if err != nil {
+		return err
+	}
+
+	// TODO(drewk): check if this is necessary.
```

Comment:
i don't see why we would want to check privileges while adding a back-ref. what was the background behind this line?